src/test/java/org/commcare/cases/test/CaseExternalIdTest.java

```diff
@@ -0,0 +1,66 @@
+package org.commcare.cases.test;
+
+import org.commcare.test.utilities.CaseTestUtils;
+import org.commcare.test.utilities.TestProfileConfiguration;
+import org.commcare.util.mocks.MockDataUtils;
+import org.commcare.util.mocks.MockUserDataSandbox;
+import org.javarosa.core.model.condition.EvaluationContext;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Collection;
+
+/**
+ * Test suite to verify end-to-end parsing of inbound case XML
+ * and reading values back from the casedb model
+ *
+ * @author ctsims
```

Comment:
author should be you.

Also: other test cases: 
1) [external_id][case_id] (dual indexes)
2) [external_id] (with no cases in the casedb)
3) [external_id and true()]